src/main/java/net/mcreator/element/GeneratableElement.java

```diff
@@ -136,6 +137,27 @@ public GeneratableElement deserialize(JsonElement jsonElement, Type type,
 				}
 
 				return generatableElement[0];
+			} catch (IllegalArgumentException e) { // we may be dealing with mod element type no longer existing

```

Comment:
When it fails, we also get a `NullPointerException`, so we need to handle it too.
After fixing this, it will be ready to be merged.
```suggestion
			} catch (IllegalArgumentException | NullPointerException e) { // we may be dealing with mod element type no longer existing

```

Reply:
What part triggers NPE?

Reply:
These are all errors I get when I open the following workspace. https://pastebin.com/0F7rPadc
(Reth is the name of the food element.)
[FullBackup202100354000.zip](https://github.com/MCreator/MCreator/files/8289873/FullBackup202100354000.zip)

Reply:
Did you remove FOOD ModElementType when testing this? In this case, line 117 should not even execute as it should fail on 115 with IllegalArgumentException

Reply:
No, FOOD MET simply does not have a specified class for the storage class (null).
![image](https://user-images.githubusercontent.com/38427877/158880972-4f29c639-e193-4ecb-a84c-9a8654a8a8f9.png)

Reply:
Hmm, you should remove it in final version, but for now, can you test this:

Add print after 

`ModElementType<?> modElementType = ModElementTypeLoader.getModElementType(newType);`

and print value of modElementType, just to check its value

Reply:
> Hmm, you should remove it in final version, but for now, can you test this:
> 
> Add print after
> 
> `ModElementType<?> modElementType = ModElementTypeLoader.getModElementType(newType);`
> 
> and print value of modElementType, just to check its value

The null or the line giving the value to the FOOD variable? If you're talking about the initialization of the variable, you put it back there.

Reply:
I did before we talked about this system.

I am wondering what value you get from ModElementTypeLoader.getModElementType(newType) for "food" string, it should throw IllegalArgumentException and catch should stop execution here, not going to line 117 in GeneratableElement

Reply:
This is everything I get.
![image](https://user-images.githubusercontent.com/38427877/158882326-7452e52c-8cd3-40ff-ae50-2e27038ff20c.png)

Reply:
When using this system (with the converter) + printing the value, I'm getting this.
![image](https://user-images.githubusercontent.com/38427877/158882806-f9de9749-503e-4fbe-ae5d-68fd44cc3cbe.png)

Reply:
Do you print string or return value of getModElementType? If is is latter and you get food, it means somehow you still register food somewhere

Reply:
> Do you print string or return value of getModElementType? If is is latter and you get food, it means somehow you still register food somewhere

I'm printing the `modElementType` variable

Reply:
Then remove this mod element type entirely, this is how this system was designed to be used, with non-existent mod elements

Reply:
> Then remove this mod element type entirely, this is how this system was designed to be used, with non-existent mod elements

Ok, I was simply applying this system to the latest commit you did and what you told me to do.

Reply:
I'm finally getting an IllegalArgumentException.

![image](https://user-images.githubusercontent.com/38427877/158885798-b0851078-71ee-4ca3-a0f8-408e9d041212.png)

Reply:
So I presume we don't need to catch NPE and PR is ok?

Reply:
> So I presume we don't need to catch NPE and PR is ok?

The PR is not ok. I'm getting an IAE using this new converter.

Reply:
Can you try with my latest change?

Reply:
It does not fix the error. I'm still getting the same error.

Reply:
I am not sure how you could get same error as now this exception is caught by my latest commit

Reply:
I can push both commits (on the Food ME PR) and you can test yourself. It may be easier for you to test everything.

Reply:
Alternatively, you can fork this PR and prepare your own solution based on the system I implemented here, just finish it to make it work

Reply:
> You can merge this PR branch into your PR and I can try it there, but I am not sure how you will be able to "unmerge" later if I do more changes

My first commit is about changes for the converter (removing the Food.class). Then, the second commit is simply adding changes of this PR, so when you are done with the PR, I can simply revert the second commit to merge this PR (so the master).

Reply:
Ok, you can do this, although maybe option 2 would be more productive and more in-line with your food to item merge work :)

Reply:
This should now be fixed, please check from your side too

Reply:
I will do it later today.